chi.go

```diff
@@ -1,81 +1,58 @@
 package chi
 
 import (
+	"context"
 	"net/http"
-
-	"golang.org/x/net/context"
 )
 
 // NewRouter returns a new Mux object that implements the Router interface.
 // It accepts an optional parent context.Context argument used by all
 // request contexts useful for signaling a server shutdown.
+
+// TODO: remove this parent context....? does it work..? do we need it?
+// or do we add our own signal to each routeContext ...?
+
 func NewRouter(parent ...context.Context) *Mux {
 	return NewMux(parent...)
 }
 
-// A Router consisting of the core routing methods used by chi's Mux.
-//
-// NOTE, the plan: hopefully once net/context makes it into the stdlib and
-// net/http supports a request context, we will remove the chi.Handler
-// interface, and the Router argument types will be http.Handler instead
-// of interface{}.
+// A Router consisting of the core routing methods used by chi's Mux,
+// built using just the stdlib.
 type Router interface {
 	http.Handler
-	Handler
-
-	Use(middlewares ...interface{})
-	Group(fn func(r Router)) Router
-	Route(pattern string, fn func(r Router)) Router
-	Mount(pattern string, handlers ...interface{})
-
-	Handle(pattern string, handlers ...interface{})
-	NotFound(h HandlerFunc)
 
-	Connect(pattern string, handlers ...interface{})
-	Head(pattern string, handlers ...interface{})
-	Get(pattern string, handlers ...interface{})
-	Post(pattern string, handlers ...interface{})
-	Put(pattern string, handlers ...interface{})
-	Patch(pattern string, handlers ...interface{})
-	Delete(pattern string, handlers ...interface{})
-	Trace(pattern string, handlers ...interface{})
-	Options(pattern string, handlers ...interface{})
+	Use(middlewares ...func(http.Handler) http.Handler)
+	Stack(fn func(r Router)) Router
+	Group(pattern string, fn func(r Router)) Router
```

Comment:
To be honest, I like the original `r.Route("/route", subrouter)` and `r.Group(subrouter)` better. Stack() sounds confusing to me, since it doesn't really trigger "custom middleware stack" in my head. And I'm not huge fan of changing .Group() logic either.